### PR TITLE
Adding idNumber and idType to OrderLookup 

### PR DESCRIPTION
**Pull request by LynnElectrum**
- Adding IdType and IdNumber query parameter
- Adding "CUSTOMER_PROFILE_ID" enum to IdType
- Feature so Minor version bump

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -38,14 +38,19 @@
     <swagger-version>1.5.9</swagger-version>
     <jetty-version>9.3.24.v20180605</jetty-version>
     <jersey-version>2.22.1</jersey-version>
-    <hibernate-validator-version>5.2.2.Final</hibernate-validator-version>
+    <hibernate-validator-version>5.4.3.Final</hibernate-validator-version>
```
Comment (MikeField119) [inline pom.xml:41]: Commenting here just because I can't above.
I don't think we need to bump the `service-interface-base-version` since we didn't change any of the models which inherit from here.
We should double check this with an API guru though.

Comment (MikeField119) [reply pom.xml:41]: @TimElectrum

**Review by MikeField119 — CHANGES_REQUESTED** [review]
Couple of small changes from my side, otherwise looks pretty good. I've asked @TimElectrum to have a look as well.
Thanks for all the effort with the testing!

File: pom.xml (hunk 1 above)
Comment (TimElectrum) [reply pom.xml:41]: it's good practice to bump the base API as you go along. Otherwise it's one day forced to go through a large version change and then a bunch unexpected issues crop up.

As much time as you may have spent on this already, encountering a few small issues as you go is much better than encountering all those same issues all at once in future.

Keep an eye out for any breaking changes which may have snuck into the base API (i.e. comments in the release notes which mention breaking changes even if the major version didn't change).

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 2]
```diff
@@ -1,5 +1,19 @@
 This page describes changes to the Money Transfer Retailer Interface implemented across different releases of the interface.
 
+## v2.25.0
+
+Released 16 September 2022
+
+- Added `idNumber` and `idType` to `lookupOrder`'s query parameters.
```
Comment (TimElectrum) [inline src/docs/devguide/hugo/site/content/release-notes.md:7]: pre-emptive comment:

when you bump the base API, please list the changes introduced by the bump here (You can basically copy/paste the release notes entries). The reason is that the base API's release notes aren't publicised so _these_ release notes would actually be the first time those changes are made public.

File: src/main/java/io/electrum/moneytransfer/model/IdType.java [hunk 3]
```diff
@@ -11,7 +11,8 @@ public enum IdType {
    PASSPORT("PASSPORT"),
    DRIVERS_LICENCE("DRIVERS_LICENCE"),
    ASYLUM_DOCUMENT("ASYLUM_DOCUMENT"),
-   UNKNOWN("UNKNOWN");
+   UNKNOWN("UNKNOWN"),
+   OTHER("OTHER");
```
Comment (TimElectrum) [inline src/main/java/io/electrum/moneytransfer/model/IdType.java:15]: what's the difference between `UNKNOWN` and `OTHER`?

The point of all of these is that they tell what the identifier is so that you can look it up correctly e.g. don't go looking in the list of national ID numbers if you've been given a passport number.

But if the `idType` is `OTHER` - well that's a pretty broad group. So broad as to be almost meaningless, no?

What does `OTHER` tell us that `UNKOWN` doesn't?

File: src/test/java/io/electrum/moneytransfer/models/NewModelTests.java [hunk 4]
```diff
@@ -0,0 +1,549 @@
+package io.electrum.moneytransfer.models;
```
Comment (TimElectrum) [inline src/test/java/io/electrum/moneytransfer/models/NewModelTests.java:1]: tbh - I haven't gone over the tests with a fine-toothed comb. You didn't really change any of the models so I'm happy enough that you didn't breaking them in this PR

File: src/test/java/io/electrum/vas/api/TestMoneyTransferApiInterfaces.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+   @DataProvider(name = "existingBaseImplementationMethodsDatProvider")
+   public Object[][] existingBaseImplementationMethodsDatProvider() {
+      return new Object[][] {
+            // the following methods all have implementations which must be used by default when overloading these
+            // methods
+            //@formatter:off
+            {IOrdersResourceTestImpl.class,"confirmPayment"},
+            {IOrdersResourceTestImpl.class,"confirmRedeem"},
+            {IOrdersResourceTestImpl.class,"createOrder"},
+            {IOrdersResourceTestImpl.class,"lookupOrder"},
```
Comment (TimElectrum) [inline src/test/java/io/electrum/vas/api/TestMoneyTransferApiInterfaces.java:22]: I wonder if you should exclude the `lookupOrder` operation/method because of its weird edge case situation

File: src/test/java/io/electrum/vas/api/TestMoneyTransferApiResourceClasses.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+   }
+
+   @DataProvider(name = "resourceInterfaceMethodsDataProvider")
+   public Object[][] resourceInterfaceMethodsDataProvider() {
+      return new Object[][] {
+         //@formatter:off
+         {OrdersResourceTestClass.class, "confirmPayment", IOrdersResource.class, "confirmPayment"},
+         {OrdersResourceTestClass.class, "confirmRedeem", IOrdersResource.class, "confirmRedeem"},
+         {OrdersResourceTestClass.class, "createOrder", IOrdersResource.class, "createOrder"},
+         {OrdersResourceTestClass.class, "lookupOrder", IOrdersResource.class, "lookupOrder"},
```
Comment (TimElectrum) [inline src/test/java/io/electrum/vas/api/TestMoneyTransferApiResourceClasses.java:32]: again, maybe this operation/method should be dropped

**Review by TimElectrum — COMMENTED** [review]
happy enough. Just some small things but I'm happy for Mike to give the okay once they're addressed. Or you're welcome to send it back to me

File: src/test/java/io/electrum/vas/api/TestMoneyTransferApiInterfaces.java (hunk 5 above)
Comment (LynnElectrum) [reply src/test/java/io/electrum/vas/api/TestMoneyTransferApiInterfaces.java:22]: the lookupOrder didnt have an edge case I believe. are you referring to getFeeQuotes?

**Review by MikeField119 — APPROVED** [review]
LGTM, I'm going to make 2 tiny updates to the README and release notes, and ask @TimElectrum for a last review and to resolve his comments.

**Review by TimElectrum — APPROVED** [review]
LGTM
